blocks/identity-block/features/signup/default.jsx

```diff
@@ -0,0 +1,173 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from '@arc-fusion/prop-types';
+import { isServerSide } from '@wpmedia/engine-theme-sdk';
+import getProperties from 'fusion:properties';
+import getTranslatedPhrases from 'fusion:intl';
+import { PrimaryFont } from '@wpmedia/shared-styles';
+import FormInputField, { FIELD_TYPES } from '../../components/FormInputField';
+import useIdentity from '../../components/Identity';
+
+import './styles.scss';
+
+function validatePasswordRegex(
+  pwLowercase,
+  pwMinLength,
+  pwPwNumbers,
+  pwSpecialCharacters,
+  pwUppercase,
+) {
+  // eslint-disable-next-line no-useless-escape
+  return `^(?=.*[a-z]{${pwLowercase},})(?=.*[A-Z]{${pwUppercase},})(?=.*\d{${pwPwNumbers},})(?=.*[@$!%*?&]{${pwSpecialCharacters},})[A-Za-z\d@$!%*?&]{${pwMinLength},}$`;
+}
```

Comment:
```suggestion
function validatePasswordRegex(
  pwLowercase,
  pwMinLength,
  pwPwNumbers,
  pwSpecialCharacters,
  pwUppercase,
) {
  // eslint-disable-next-line no-useless-escape
  const expression = [
    ...(pwLowercase > 0 ? [`(?=.*[a-z]{${pwLowercase},})`] : []),
    ...(pwUppercase > 0 ? [`(?=.*[A-Z]{${pwUppercase},})`] : []),
    ...(pwPwNumbers > 0 ? [`(?=.*\d{${pwPwNumbers},})`] : []),
    ...(pwSpecialCharacters > 0 ? [`(?=.*[@$!%*?&]{${pwSpecialCharacters},})`] : []),
    ...(pwMinLength > 0 ? [`[A-Za-z\d@$!%*?&]{${pwMinLength},}`] : ['[A-Za-z\d@$!%*?&]*']),
  ].join();
  return `^${expression}$`;
}
```

Reply:
Would this be a better regexp combiner to rule out the 0 values?

Reply:
I just realized that the lowercase, uppercase, number, and special rules are basically useless since the minlength rule at the end will just accept all of them anyway.

Reply:
oooh thank you -- this looks good

Reply:
will test it out tomorrow and validate

Reply:
I edited the code block to properly join the password mask:
```
  ].join('');  /*<--- this line
```

Reply:
resolved this with //d